dotnet/src/webdriver/Response.cs

```diff
@@ -56,59 +52,65 @@ public Response(SessionId sessionId)
         {
             if (sessionId != null)
             {
-                this.responseSessionId = sessionId.ToString();
+                this.SessionId = sessionId.ToString();
             }
         }
 
+        internal Response(SessionId sessionId, WebDriverResult status, object value)
```

Comment:
Ideally the parameterless constructor `Response()` can be completely removed. It sets all values as default (keeping in mind `default(WebDriverResult)` is `Success`). There is one last place where the parameterless ctor is used, but it's less trivial to unwind.

Unfortunately, the parameterless constructor (and `Response(SessionId)` which has the same problem) is `public` so it may have to stay. But it should be obsolete (and if users complain, we can make this `internal` into `public`.

Reply:
All 3 properties have public setters. Is it possible to make those obsolete? This class ideally should be immutable, and those setters were used in code such as you see me changing in this PR in `WebDriver`. Better served by setting in ctor + immutable once created.

Reply:
Right, this class should be immutable. Let's try to do it, and double check whether Appium will not be broken.

In case of success, then just rename this PR with `[breaking change]` prefix.

Reply:
This PR does a bunch of things, but does not include the breaking change I'm referring to. It would first require big changes to `HttpCommandExecutor.CreateResponse`.

This PR does not have any breaking changes.

Reply:
Discussion no longer relevant to this PR. I stripped it down to its core.